### ISSUE TITLE
requires a Cargo.lock file

### ISSUE THREAD

**Issue by mimoo** [CircleCI-Public/rust-orb#18]
There are errors if you don't have a Cargo.lock file, which is common if your repo is a library.

**Comment by gmemstr** [comment]
Yep, this is on me. It was only recently that I realized the use case for the Cargo.lock file (given my knowledge of Node/npm/yarn, I assumed it was beneficial in all cases to have it committed). @mimoo Do you believe relying only on the Cargo.toml file is enough? Or should we try to support both through a parameter.

**Comment by mimoo** [comment]
Is the Cargo.lock file used for caching? In that case yeah I think relying on Cargo.toml could be a good idea, it's just that the cache is going to become invalidated more often (with each patch version increase, for example).

**Comment by gmemstr** [comment]
Correct - it's used as a primary part of the cache key. But if it's not committed the cache will never be loaded.

**Comment by mimoo** [comment]
Interesting, I think a better way to handle that in Rust would be to just use the latest cache used to build master

**Comment by marboledacci** [comment]
I see this issue is old, please update if this is still required and provide more details, or close it.
